apis/v1alpha1/nginxproxy_types.go

```diff
@@ -0,0 +1,81 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+// +kubebuilder:resource:categories=nginx-gateway-fabric,scope=Cluster
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+
+// NginxProxy is a configuration object that is attached to a GatewayClass parametersRef. It provides a way
+// to configure global settings for all Gateways defined from the GatewayClass.
+type NginxProxy struct { //nolint:govet // standard field alignment, don't change it
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of the NginxProxy.
+	Spec NginxProxySpec `json:"spec"`
+}
+
+// +kubebuilder:object:root=true
+
+// NginxProxyList contains a list of NginxProxies.
+type NginxProxyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []NginxProxy `json:"items"`
+}
+
+// NginxProxySpec defines the desired state of the NginxProxy.
+type NginxProxySpec struct {
+	// Telemetry specifies the OpenTelemetry configuration.
+	//
+	// +optional
+	Telemetry *Telemetry `json:"telemetry,omitempty"`
+}
+
+// Telemetry specifies the OpenTelemetry configuration.
+type Telemetry struct {
+	// Exporter specifies OpenTelemetry export parameters.
+	//
+	// +optional
+	Exporter *TelemetryExporter `json:"exporter,omitempty"`
+
+	// ServiceName is the "service.name" attribute of the OpenTelemetry resource.
+	// Default is 'ngf:<gateway-namespace>:<gateway-name>'. If a value is provided by the user,
+	// then the default becomes a prefix to that value.
+	//
+	// +optional
+	// +kubebuilder:validation:MaxLength=127
```

Comment:
This length can be up for debate. Standard max length for things like this is 255, however since we are prefixing with a string, I wanted to reduce this max length to account for that. We could still technically go over 255, but I don't think nginx or OTEL have any limits on this length (verification while implementing may unveil this, however).